homeassistant/components/roon/event.py

```diff
@@ -0,0 +1,84 @@
+"""Roon event entities."""
+import logging
+
+from homeassistant.components.event import EventDeviceClass, EventEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Roon Event from Config Entry."""
+    roon_server = hass.data[DOMAIN][config_entry.entry_id]
+    event_entities = set()
+
+    @callback
+    def async_add_roon_volume_entity(player_data):
+        """Add or update Roon event Entity."""
+        dev_id = player_data["dev_id"]
+        if dev_id in event_entities:
+            return
+        # new player!
+        event_entity = RoonEventEntity(roon_server, player_data["display_name"])
+        event_entities.add(dev_id)
+        async_add_entities([event_entity])
+
+    # start listening for players to be added from the server component
+    async_dispatcher_connect(hass, "roon_media_player", async_add_roon_volume_entity)
+
+
+class RoonEventEntity(EventEntity):
+    """Representation of a Roon Event entity."""
+
+    _attr_device_class = EventDeviceClass.BUTTON
+    _attr_event_types = ["volume_up", "volume_down"]
+
+    def __init__(self, server, name):
+        """Initialize the entity."""
```

Comment:
Connect this entity to the same Home Assistant device as the media player by setting `device_info`.

https://developers.home-assistant.io/docs/device_registry_index#defining-devices